source/opt/types.cpp

```diff
@@ -455,12 +455,12 @@ std::string Struct::str() const {
 }
 
 void Struct::GetExtraHashWords(std::vector<uint32_t>* words) const {
-  for (auto t : element_types_) {
+  for (auto* t : element_types_) {
```

Comment:
auto and auto* don't have much difference, except for more type safety.

Reply:
Yes.  After seeing #1260 I went through and annotated things more explicitly.  Having the * there makes it clearer this is an inexpensive iteration.   So it helps verifiability.